service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/registry/RegistryIdentity.java

```diff
@@ -275,20 +275,51 @@ else if (this.authentication.getAuthenticationType() == AuthenticationType.SAS)
      */
     RegistryIdentity(RegistryIdentityParser parser) throws IllegalArgumentException
     {
-        if (parser.getAuthenticationParser() == null || parser.getAuthenticationParser().getType() == null)
```

Comment:
I find that handling error cases immediately makes for a better reading experience. Otherwise one must scroll down a ways, try to figure out the indenting level, and find the relevant exception associated with these being null.

Reply:
I can reorder this a bit, but this is no longer an error case. It'll be something like

```java
//do all other parsing here, saving authentication type for last

if (parser.getAuthenticationParser() == null || parser.getAuthenticationParser().getType() == null)
{
    return;
}

// assign the authentication type here
```